pkg/resources/ratelimit/envoyconfig.go

```diff
@@ -0,0 +1,50 @@
+package ratelimit
+
+import (
+	"context"
+
+	marin3rv1alpha1 "github.com/3scale/marin3r/pkg/apis/marin3r/v1alpha1"
+	integreatlyv1alpha1 "github.com/integr8ly/integreatly-operator/pkg/apis/integreatly/v1alpha1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func DeleteEnvoyConfigsInNamespaces(ctx context.Context, client k8sclient.Client, namespaces ...string) (integreatlyv1alpha1.StatusPhase, error) {
+	phase := integreatlyv1alpha1.PhaseCompleted
+
+	for _, namespace := range namespaces {
+		nsPhase, err := DeleteEnvoyConfigsInNamespace(ctx, client, namespace)
+		if err != nil {
+			return integreatlyv1alpha1.PhaseFailed, err
+		}
+
+		// Only change the status phase if it was Completed, to ensure that
+		// as long as one of the namespaces returns InProgress, the phase is
+		// set to InProgress
+		if phase == integreatlyv1alpha1.PhaseCompleted {
+			phase = nsPhase
+		}
+	}
+
+	return phase, nil
```

Comment:
Not quite sure I understand this. `phase` is already preset to `PhaseComplete` so this will always be true for the first run of the for loop. So is the thinking, `phase` gets set to whatever `nsPhase` is and be unblocking for each namespace. Then on the next reconcile, the `phase` will only be complete if all the `EnvoyConfigs` are deleted for each namespace  :thinking: 

Is there any problems with just a return if `nsPhase` is `PhaseInProgress` instead? Might be a bit easier to understand :thinking: 

```suggestion
	phase := integreatlyv1alpha1.PhaseCompleted

	for _, namespace := range namespaces {
		nsPhase, err := DeleteEnvoyConfigsInNamespace(ctx, client, namespace)
		if err != nil {
			return integreatlyv1alpha1.PhaseFailed, err
		}

		if phase == integreatlyv1alpha1.InProgress {
			return nsPhase, err 
		}
	}

	return phase, nil
```

Reply:
This seems like a good point to me, thanks @KevFan, wdyt @sergioifg94

Reply:
Actually might be able to use the same 2 checks from the previous comments here instead also :thinking: 
```suggestion
		for _, namespace := range namespaces {
		nsPhase, err := DeleteEnvoyConfigsInNamespace(ctx, client, namespace)
		if err != nil || nsPhase != integreatlyv1alpha1.PhaseCompleted {
			return nsPhase, err
		}
	}
	return integreatlyv1alpha1.PhaseCompleted, nil
```

Reply:
@KevFan `phase` is set to whatever `nsPhase` is but only if it currently is PhaseComplete. If it is InProgress it will always stay in progress. Your suggestion would fulfill that purpose, but in the scenario that the first iteration of the loop returns `InProgress`, the whole function would be shortcut and it wouldn't reach the other namespaces. This approach ensures that at least we send the Delete request for all namespaces no matter the order

Reply:
@sergioifg94 Thanks for the explaination. Yeah fair, I'm good either or anyway, more of a question than anything :smiley: . I think the main fix is there and currently only 1 namespace is passed to this function anyhow :+1:

Reply:
Oh yes I forgot to mention that. For now it's only one namespace so it doesn't make a difference either way :stuck_out_tongue: